packages/table/src/useLeafyGreenTable/TableCheckbox.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Row, Table } from '@tanstack/react-table';
+
+import Checkbox from '@leafygreen-ui/checkbox';
+import { cx } from '@leafygreen-ui/emotion';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+
+import { useRowContext } from '../Row/RowContext';
+
+import { disabledTableRowCheckStyles } from './useLeafyGreenTable.styles';
+import { LGRowData, LGTableDataType } from '.';
+
+export const TableHeaderCheckbox = ({
+  table,
+}: {
+  table: Table<LGTableDataType<LGRowData>>;
+}) => {
+  const { theme } = useDarkMode();
+  const { disabled: rowIsDisabled } = useRowContext();
+  return (
+    <Checkbox
+      className={cx({
+        [disabledTableRowCheckStyles[theme]]: rowIsDisabled,
```

Comment:
nit: i don't think you even need to conditionally render these styles with `rowIsDisabled` since you're targeting `[aria-checked='true']`.